### PR TITLE
[ET-VK] Documentation for Vulkan Delegate

### PR DESCRIPTION
**Pull request by SS-JIA**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3113

Differential Revision: [D56279743](https://our.internmc.facebook.com/intern/diff/D56279743)

### CODE REVIEW

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56279743).

**Comment by SS-JIA** [comment]
@SS-JIA has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56279743).

File: backends/vulkan/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+The easiest way to build and test the Vulkan Delegate is to build for Android
+and test on a local Android device. Android devices have built in support for
+Vulkan, and the Android NDK ships with a GLSL compiler, which is needed to
+compile the Vulkan Compute Library's GLSL compute shaders.
+
+The Vulkan Delegate libraries can be built by setting `-DEXECUTORCH_BUILD_VULKAN=ON`
+when building with CMake.
+
+First, make sure that you have the Android NDK installed - Android NDK r25c is
```
Comment (cccclai) [inline backends/vulkan/README.md:140]: Is `r25c1 required or the one tested?

**Review by cccclai — APPROVED** [review]
Looks great!

File: backends/vulkan/README.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
+# ExecuTorch Vulkan Delegate
+
+The ExecuTorch Vulkan delegate is a native GPU delegate for ExecuTorch that is
```
Comment (cccclai) [inline backends/vulkan/README.md:3]: Is there a session to describe the supported platform for both AOT and runtime?

File: backends/vulkan/README.md (hunk 1 above)
Comment (SS-JIA) [reply backends/vulkan/README.md:140]: Just recommended. I mention it to be consistent with Hansong's Android app demo.

I don't think there is a hard NDK requirement except for being after when Vulkan was added (I think r17). I tested with r19 and it was able to build too.

**Pull request closed without merging**